shanoir-ng-datasets/src/main/java/org/shanoir/ng/examination/controler/ExaminationApiController.java

```diff
@@ -260,8 +270,46 @@ public ResponseEntity<Void> addExtraData(
 		if (examinationService.addExtraData(examinationId, file) != null) {
 			return new ResponseEntity<>(HttpStatus.OK);
 		}
-		return new ResponseEntity<>(HttpStatus.NOT_ACCEPTABLE);
+		return new ResponseEntity<>(HttpStatus.UNPROCESSABLE_ENTITY);
 	}
+	
+	@Override
+	public ResponseEntity<Void> createExaminationAndAddExtraData(
+			@ApiParam(value = "name of the subject", required = true) @PathVariable("subjectName") String subjectName,
+			@ApiParam(value = "id of the center", required = true) @PathVariable("centerId") Long centerId,
+			@ApiParam(value = "file to upload", required = true) @Valid @RequestBody MultipartFile file) throws RestServiceException {
+		
+		Subject subject = subjectRepository.findByName(subjectName);
+		if (subject == null) {
+			ErrorModel error = new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(), "Couldn't find subject with name " + subjectName);
+			throw new RestServiceException(error);
+		}
+		
+		if (centerRepository.findById(centerId).isEmpty()) {
+			ErrorModel error = new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(), "Couldn't find center with id " + centerId);
+			throw new RestServiceException(error);
+		}
+
+		Examination examination = new Examination();
+		examination.setComment(file.getOriginalFilename());
+		examination.setCenterId(centerId);
+		examination.setSubjectId(subject.getId());
+		examination.setStudyId(subject.getSubjectStudyList().get(0).getStudy().getId());
```

Comment:
I guess that in the specifications, all the subjects are only part of one study ?